manifests/init.pp

```diff
@@ -51,6 +54,24 @@
     }].to_yaml %>')
   $policies_data = parseyaml($policies_yaml)
 
+  if $::rh_certificate_repo_ca_file {
```

Comment:
Should there maybe be some module-level toggle here? 

I think you'll get a value for these RH facts on any RHEL system with subscription-manager installed, and because it's included in the @Base installation is probably most systems.

Alternatively, you can check ONLY the CA file, and that the fact value is not specified or is the default (braketed, [/etc/rhsm/ca/redhat-uep.pem]).  The default would indicate the system isn't subscribed to a Katello, because it's pointing at Red Hat CDN.